examples/faq/are_you_there.py

```diff
@@ -0,0 +1,65 @@
+from confluent_kafka import Producer, KafkaError
+
+
+class ConfluentException(Exception):
+    """
+        Converts KafkaError into an exception for handling by an application.
+
+        :param KafkaError kafka_error: The KafkaError to convert.
+        :param str msg: Optional Error message override.
+    """
+    def __init__(self, kafka_error, msg=None):
+        self.code = kafka_error.code()
+
+        if msg is None:
+            msg = kafka_error.str()
+
+        self.message = msg
+
+
+class DeliveryReporter(object):
+    __slots__ = ["success"]
+
+    def __init__(self):
+        self.success = False
+
+    def __call__(self, err, msg):
+        """
+        Handles per message delivery reports indicating success or failed delivery attempts.
+
+        :param err: KafkaError indicating message delivery failed.
+        :param msg: The message delivered to the broker.
+        """
+        if err:
+            self.error(err)
+
+        self.success = True
+        print("Message {} delivered!".format(msg.offset()))
+
+    def error(self, err):
+        """
+        Handler for global/generic and message delivery errors.
+
+        Raises ConfluentException for `KafkaError.__ALL_BROKERS_DOWN` if seen before a successful delivery.
+        Often times this is the result of a misconfiguration.
+        """
+        if not self.success and err.code() is KafkaError._ALL_BROKERS_DOWN:
```

Comment:
This is problematic, all brokers could go down at any time for whatever reason and be back up the next second, librdkafk has been carefully crafted not to make this a fatal error, but instead abstract the required retries, restoration, etc. The approach to error handling layed out here voids all that, at no benefit.

This is indeed a big and complex subject, but I think we need to all agree on the how,when and what of error propagation before sending our users down any specific error handling path.

E.g., a common ask is to have a way for an application to quickly know if its broker list, security settings, are erroneously configured, ALL_BROKERS_DOWN is suitable for this, but only at startup.
IF an application has been running successfully for 3 weeks and suddenly gets an ALL_BROKERS_DOWN it is probably not due to configuration error.
How do we differentiate? That's the big question,

Reply:
Understood, that's why I'm offering an example of something one might do as opposed to trying to put it in the client. This example is not intending to solve distributed computing rather it  provides an example of how one might notify the application that a particular type of error occured. In this case that error is ALL_BROKERS_DOWN in the absence of any proof a connection (successful send). Raising exceptions under certain error conditions so the application can decide what to do doesn't seem terribly misguided to me but maybe I'm wrong. 

What if we include a healthy disclaimer in the example documentation and a fuller example where we wrap our main producer logic in a try.